null database strategy to disable @javascript handling

akostadinov opened the pull request:
<!-- NAMING YOUR PULL REQUEST: Please choose a concise, descriptive name for your pull request. -->
<!-- This makes it easier to get some context when reading the names of issues -->

<!-- These sections are meant as guidance for you. If something doesn't fit, you can just skip it. -->

## Summary

Presently there is no nice way to disable `@javascript` tag handling without writing a custom database strategy class. See #521

This PR adds a null database strategy to allow easily doing so by

```
Cucumber::Rails::Database.javascript_strategy = :none
```

I'm planning to add some other enhancements and documentation later to allow custom list of tags but having a null strategy is useful anyway and is a first step.

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

aurelien-reeves commented:
What about naming it "none" instead of "null"?

akostadinov commented:
@aurelien-reeves , I agree with whatever naming project wants. If the change is generally accepted, I'd go ahead to rename it.

akostadinov commented:
I renamed to none.

aurelien-reeves reviewed (commented):
It looks good to me, well done 👌
But as I am not that much into this project, I prefer to let other contributors to review it :)

aurelien-reeves commented:
@akostadinov could you add an entry in the changelog? And if you had the opportunity to add some documentation regarding those database strategies, that would be a good bonus to that PR :)

@olleolleolle @luke-hill if no further objection, I may merge this soon

luke-hill reviewed (changes requested):
This feels like it's not a None strategy i.e. one isn't there, but a null strategy. So perhaps we should name it NullStrategy to infer that the responses on the js checkers are nil

olleolleolle reviewed (approved):
Looks good to me.

akostadinov commented:
Due to popular demand, I renamed class back to `NullStrategy`. And `:none` as strategy selection makes sense to me to indicate that we use none of the javascript testing strategies. `:null` also makes sense to be consistent with the class name though. Probably `none` reads a little better. I have arguments in favor of both options so I can change to whatever project owners want.

@luke-hill , added changelog and updated doc to clarify `none` and `deletion` are symbols. I have some other modifications ready that can make the feature useful for whatever tags a project decides to use. I think it's better to submit changes in steps and this is the first step. I want to add information to the readme once I have a review and some agreement about the other changes. Is this ok?

akostadinov commented:
Guys, can we merge in case of no objections?
